lab3/client/src/reducer.js

```diff
@@ -0,0 +1,37 @@
+export default function reducer (state, action) {
+	switch (action.type) {
+	case 'EXAMPLE_MUTATION':
+		state.example = action.payload;
+		return state;
+
+	case 'BUY_GENERATOR':
+
+
+		for (var i = 0; i < state.generators.length; i++) {
+
+			if (action.payload.name == state.generators[i].name) {
+				var rt = Math.pow(1+0.05, state.generators[i].quantity);
+				var xt = state.generators[i].baseCost*(rt);
```

Comment:
should probably just resuse the implementation from the generator.getCost. But I get what you meant.